src/main/java/org/llorllale/cactoos/matchers/HasValues.java

```diff
@@ -26,72 +26,72 @@
  */
 package org.llorllale.cactoos.matchers;
 
-import org.cactoos.Func;
-import org.cactoos.func.UncheckedFunc;
-import org.cactoos.list.ListOf;
-import org.cactoos.scalar.Or;
-import org.cactoos.scalar.UncheckedScalar;
+import java.util.Collection;
+import org.cactoos.collection.CollectionOf;
+import org.cactoos.iterable.IterableOf;
 import org.hamcrest.Description;
 import org.hamcrest.TypeSafeDiagnosingMatcher;
 
 /**
- * Matcher to check the {@link Iterable} by particular function.
+ * Matcher to check that {@link Iterable} has particular elements.
  *
- * <p>Here is a few examples how {@link HasValues} can be used:</p>
+ * <p>Here is an example how {@link HasValues} can be used:</p>
  * <pre>
  *  MatcherAssert.assertThat(
- *    new ListOf<>(2, 3, 4),
- *    new HasValues<>(val -> val > 1)
- *  );
- *  MatcherAssert.assertThat(
- *    new ListOf<>("line A", "line B", "line C"),
- *    new HasValues<>(val -> val.startsWith("line"))
- *  );</pre>
+ *     new ListOf<>(1, 2, 3),
+ *     new HasValues<>(2)
+ * );</pre>
  *
  * @param <X> Type of item.
  * @since 1.0.0
+ * @todo #36:30min HasValuesMatching that accepts matching function.
+ *  For example, the "non-null", "startWith", "endsWith", "greaterThan" etc.
+ *  The HasValue matcher should work only with pre-defined set of expected items
  * @checkstyle ProtectedMethodInFinalClassCheck (200 lines)
  */
 public final class HasValues<X> extends TypeSafeDiagnosingMatcher<Iterable<X>> {
+
     /**
-     * The function to check the {@link Iterable}.
+     * The expected values within the collection.
      */
-    private final UncheckedFunc<X, Boolean> fnc;
+    private final Collection<X> expected;
```

Comment:
@dgroup we frown upon the use of `Collection`, preferring `Iterable` instead

Reply:
@llorllale done.